qendpoint-core/src/main/java/com/the_qa_company/qendpoint/core/triples/impl/BitmapTriplesIterator.java

```diff
@@ -24,11 +24,12 @@
 import com.the_qa_company.qendpoint.core.iterator.SuppliableIteratorTripleID;
 import com.the_qa_company.qendpoint.core.triples.TripleID;
 import com.the_qa_company.qendpoint.core.compact.bitmap.AdjacencyList;
+import org.eclipse.rdf4j.common.iteration.IndexReportingIterator;
 
 /**
  * @author mario.arias
  */
-public class BitmapTriplesIterator implements SuppliableIteratorTripleID {
+public class BitmapTriplesIterator implements SuppliableIteratorTripleID, IndexReportingIterator {
```

Comment:
I've added a new interface to RDF4J which allows the query explanation to include the name of the index used in StatementPattern.